hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/TimesTenDialect.java

```diff
@@ -157,21 +177,133 @@ public int getDefaultDecimalPrecision() {
+		CommonFunctionFactory functionFactory     = new CommonFunctionFactory(functionContributions);
+		final BasicTypeRegistry basicTypeRegistry = typeConfiguration.getBasicTypeRegistry();
+		final BasicType<Date>   timestampType     = basicTypeRegistry.resolve( StandardBasicTypes.TIMESTAMP );
+		final BasicType<Date>   dateType          = basicTypeRegistry.resolve( StandardBasicTypes.DATE );
+		final BasicType<Date>   timeType          = basicTypeRegistry.resolve( StandardBasicTypes.TIME );
+		final BasicType<String> stringType        = basicTypeRegistry.resolve( StandardBasicTypes.STRING );
+		final BasicType<Long>   longType          = basicTypeRegistry.resolve( StandardBasicTypes.LONG );
+		final BasicType<Integer>intType           = basicTypeRegistry.resolve( StandardBasicTypes.INTEGER );
+
+		// String Functions
+		functionContributions.getFunctionRegistry().register( 
+				"rtrim", new StandardSQLFunction("rtrim", StandardBasicTypes.STRING) 
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"ltrim", new StandardSQLFunction("ltrim", StandardBasicTypes.STRING) 
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"length", new StandardSQLFunction("length", StandardBasicTypes.LONG)
+		);
+		functionFactory.concat_pipeOperator();
+		functionContributions.getFunctionRegistry().register( 
+				"to_char", new StandardSQLFunction("to_char", StandardBasicTypes.STRING)
+		);
+		functionFactory.char_chr();
 		functionFactory.instr();
 		functionFactory.substr();
-		functionFactory.substring_substr();
-		functionFactory.leftRight_substr();
-		functionFactory.char_chr();
-		functionFactory.rownumRowid();
-		functionFactory.sysdate();
-		functionFactory.addMonths();
-		functionFactory.monthsBetween();
+		functionContributions.getFunctionRegistry().register( 
+				"str", new StandardSQLFunction("to_char", StandardBasicTypes.STRING)
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"substring", new StandardSQLFunction( "substr", StandardBasicTypes.STRING )
+		);
+		functionFactory.soundex();
+
+		// Date/Time Functions
+		functionContributions.getFunctionRegistry().register( 
+				"to_date", new StandardSQLFunction("to_date", StandardBasicTypes.TIMESTAMP)
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"sysdate", new CurrentFunction("sysdate", "sysdate", timestampType)
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"getdate", new StandardSQLFunction("getdate", StandardBasicTypes.TIMESTAMP)
+		);
+
+		functionContributions.getFunctionRegistry().register( 
+				"current_date",      new CurrentFunction("sysdate", "sysdate", dateType) 
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"current_time",      new CurrentFunction("sysdate", "sysdate", timeType) 
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"current_timestamp", new CurrentFunction("sysdate", "sysdate", timestampType) 
+		);
```

Comment:
Already covered by `Dialect.initializeFunctionRegistry`, which uses the `currentTime`, `currentTimestamp` and `currentDate` of this `Dialect` to determine the SQL fragment.



```suggestion
```